src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,216 @@
+#include "postgres.h"
+
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#if PG_VERSION_NUM >= 120000
+#include "access/table.h"
+#endif
+#include "catalog/pg_authid.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/deparser.h"
+#include "distributed/master_protocol.h"
+#include "distributed/worker_transaction.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+#include "utils/rel.h"
+#include "utils/syscache.h"
+#include "nodes/makefuncs.h"
+
+List * ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString);
+List * CreateAlterRoleQueriesForCopyingAllUsers(void);
+void PutEncryptedPasswordInQueryTree(DefElem *option, const char *rolename);
+const char * prepareAlterRoleQueryWithUDF(AlterRoleStmt *stmt);
+
+
+bool EnableAlterRolePropagation = true;
+
+/*
+ * ProcessAlterRoleStmt actually creates the plan we need to execute for alter
+ * role statement.
+ */
+List *
+ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString)
+{
+	ListCell *optionCell = NULL;
+	List *commands = NIL;
+
+	if (!IsCoordinator())
+	{
+		return NIL;
+	}
+
+	foreach(optionCell, stmt->options)
+	{
+		DefElem *option = (DefElem *) lfirst(optionCell);
+
+		if (EnableAlterRolePropagation && strcasecmp(option->defname, "password") == 0)
```

Comment:
we should now pull `EnableAlterRolePropagation ` to above, before the loop starts

Reply:
Done 👍